File: app/src/main/java/org/deiverbum/app/data/entity/MassReadingWithAll.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.relation.BibleReadingWithBook;
import org.deiverbum.app.data.entity.relation.LiturgyWithTime;
import org.deiverbum.app.data.entity.relation.MassReadingWithBook;
import org.deiverbum.app.model.MassReading;

import java.util.List;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
public class MassReadingWithAll {
    @Embedded
    /*public LiturgyGroupEntity groupEntity;

    @Relation(
            parentColumn = "groupID",
            entityColumn = "groupFK",
            entity = MassReadingEntity.class
    )
*/
    public MassReadingEntity massReadingEntity;


    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingID",
            entity = BibleReadingEntity.class
    )
    public BibleReadingWithBook lectura;

    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingID",
            entity = BibleReadingEntity.class
    )
    public List<BibleReadingEntity> lecturass;


    @Relation(
            parentColumn = "readingFK",
            entityColumn = "readingID",
            entity = BibleReadingEntity.class
    )
    public List<MassReadingWithBook> lecturas;


    @Relation(
            parentColumn = "liturgyFK",
            entityColumn = "liturgyID",
            entity = LiturgyEntity.class
    )
    public LiturgyWithTime liturgia;

    public MassReading getDomainModel() {
        //MassReading theModel=new MassReading();
        MassReading theModel = lectura.getDomainModelMisa();
        theModel.setReadingID(massReadingEntity.readingFK);
        theModel.setTema(massReadingEntity.getTema());
        theModel.setOrden(massReadingEntity.getOrden());
        return theModel;
    }


}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/MassReadingWithAll.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.BibleReadingEntity
import org.deiverbum.app.data.entity.MassReadingEntity
import org.deiverbum.app.model.MassReading

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
class MassReadingWithAll {
    @JvmField
    @Embedded
    var massReadingEntity: MassReadingEntity? = null

    @JvmField
    @Relation(
        parentColumn = "readingFK",
        entityColumn = "readingID",
        entity = BibleReadingEntity::class
    )
    var lectura: BibleReadingWithBook? = null
    val domainModel: MassReading
        get() {
            val theModel = lectura?.domainModelMisa
            theModel!!.readingID = massReadingEntity!!.readingFK
            theModel.tema = massReadingEntity!!.tema
            theModel.setOrden(massReadingEntity!!.orden)
            return theModel
        }
}